qiskit/circuit/instruction.py

```diff
@@ -220,24 +220,49 @@ def assemble(self):
             instruction._condition = self.condition
         return instruction
 
-    def mirror(self):
+    def reverse(self):
```

Comment:
You'll want to keep and deprecate `mirror()` here too, since it's been a method on instruction for several releases people might be using it directly and this will break that